src/controllers/cartController.ts

```diff
@@ -1,144 +1,206 @@
+    }
+
+    const cartItems = [];
+
+    for (const cart of carts) {
+      const productId: string = cart.get('product_id') as string;
+      const product = await ProductsModel.findOne({ where: { id: productId } });
+
+      if (!product) {
+        cartItems.push({
+          ...cart.get(),
+          error: 'Product not found for the specified product ID',
+        });
       } else {
-        const product = await ProductsModel.findOne({ where: { id: product_id } });
-  
-        if (!product) {
-          return res.status(404).json({ error: 'Product not found for the specified product ID' });
-        }
-  
-        const product_name: string = product.get('name') as string;
+        const productDescription: string = product.get('description') as string;
+        const productImageSecureUrl: string = product.get('image_secure_url') as string;
         const finalPrice: number = parseFloat(product.get('finalPrice') as string);
-  
-        cart = await CartsModel.create({
-          user_id: userId,
-          product_id,
-          product_name,
+
+        cartItems.push({
+          ...cart.get(),
           finalPrice,
-          quantity,
+          productDescription,
+          productImageSecureUrl,
         });
       }
-  
-      return res.status(200).json({ message: 'Success', cart });
-    } catch (error) {
-      console.error(error);
-      return res.status(500).json({ error: 'Internal server error' });
     }
-  };
-  
-
-  export const deleteFromCart = async (req: Request, res: Response) => {
-    try {
-      const { userId, productId } = req.body;
-  
-      if (!userId || !productId) {
-        return res.status(400).json({ error: 'Both userId and productId are required in the request body' });
+
+    return res.status(200).json({ message: 'Success', cartItems });
+  } catch (error) {
+    console.error(error);
+    return res.status(500).json({ error: 'Internal server error' });
+  }
+};
+
+
+export const addToCart = async (req: Request, res: Response) => {
+  try {
+    const { error } = cartsSchema.validate(req.body);
+
+    if (error) {
+      return res.status(400).json({ error: error.details[0].message });
+    }
+
+    const { userId, quantity, product_id } = req.body;
+
+    let cart = await CartsModel.findOne({ where: { user_id: userId, product_id } });
+
+    if (cart) {
+      const currentQuantity: number = cart.get('quantity') as number;
+
+      if (quantity >= currentQuantity) {
+        await cart.update({ quantity });
       }
-  
-      const cart = await CartsModel.findOne({
-        where: { user_id: userId, product_id: productId },
-      });
-  
-      if (!cart) {
-        return res.status(404).json({ error: 'Product not found in the user\'s cart' });
+    } else {
+      const product = await ProductsModel.findOne({ where: { id: product_id } });
+
+      if (!product) {
+        return res.status(404).json({ error: 'Product not found for the specified product ID' });
```

Comment:
ERROR 404 indicates page not found, when there is item that is not found return status 204